### ISSUE TITLE
I got no beep on twitch popout tab

### ISSUE THREAD

**Issue by Ramonlaw** [steveseguin/social_stream#305]
I got no beep on twitch popout window. it just works on my youtube popout window

**Comment by steveseguin** [comment]
![image](https://github.com/user-attachments/assets/b7b2688d-7356-4013-b407-c611ff783fb0)

The twitch / youtube pop out windows don't beep, but the dock or featured page can beep on new messages, when enabled.

You also have to touch the dock/featured page before it will play sounds, so it may not always work after a page reload.

I dont see why the dock page would treat any message any different; whether from twitch or youtube, so I'd need more information to help at this point

**Comment by Ramonlaw** [comment]
![image](https://github.com/user-attachments/assets/8b793d0d-423e-47b4-b811-d018c6b565af)

my config is this but only youtube window have beep on messsages but the twitch chat have no beep on messages

**Comment by steveseguin** [comment]
I'm not able to reproduce. Twitch messages are making a beep sound with those settings, via the dock.html page. 

Are there other settings? other differences between twitch and Youtube I'm not aware of?

[I moved the "beep" function higher up in the code,](https://github.com/steveseguin/social_stream/commit/33a8d0f6109c2ec2bf8aea269c47c8d82620ad80) before the message appears. Might help with debugging or fix the issue if there was some error in the code further down the page.

Just refresh the dock page I guess, click anywhere on it, and try again

**Comment by Ramonlaw** [comment]
![image](https://github.com/user-attachments/assets/48c3aafe-1904-4a11-aca6-5d4725a5c65e)
 the left window is the twitch chat that does not beep and the right one is from youtube chat that works perfect
 
 
![image](https://github.com/user-attachments/assets/0d7c3da5-9e5b-48a3-8185-d0a8e65d1eb1)
 when i type on the obs window it shows on both and beep but on the obs i dont know witch window reproduce the beep

**Comment by Ramonlaw** [comment]
![image](https://github.com/user-attachments/assets/2dc8962e-96ed-4c16-84b2-1a0b3c8e6572)
even with this only the youtube window beep

**Comment by steveseguin** [comment]
In your screen shots, I only see messages from YouTube in the dock.  I do not see Twitch messages.

So is the issue that the beep isn't working, or is the issue that Twitch messages are not being captured?

If twitch messages are not being captured, that's a completely different issue. If refreshing the twitch pop out doesn't help fix it, it might be an incompatibility.  Maybe a language setting or other browser extension is causing an issue.

I can review the code to see if this is the case, although any details about what browser extensions you are using or any changes you made to the twitch chat pop out page.

Kindly.
steve

**Comment by Ramonlaw** [comment]
> In your screen shots, I only see messages from YouTube in the dock. I do not see Twitch messages.
> 
> So is the issue that the beep isn't working, or is the issue that Twitch messages are not being captured?
> 
> If twitch messages are not being captured, that's a completely different issue. If refreshing the twitch pop out doesn't help fix it, it might be an incompatibility. Maybe a language setting or other browser extension is causing an issue.
> 
> I can review the code to see if this is the case, although any details about what browser extensions you are using or any changes you made to the twitch chat pop out page.
> 
> Kindly. steve

yeah i know that on dock only shows youtube but it sends the message to both tabs on chrome

**Comment by steveseguin** [comment]
It won't beep unless it shows in the dock also.

If you are blocking Twitch, then this makes sense. If it's not working
though, as expected, then I'll need to figure out why it's not capturing

**Comment by Ramonlaw** [comment]
If I am blocking Twitch it was definitely not intentional because I don't even know how to block

**Comment by steveseguin** [comment]
I pushed an update for Twitch about an hour ago.  You can see if it helps to resolve the issues.  If that doesn't help, I may need to send you a custom version with some debug log output, to help me see what's going on.
![image](https://github.com/user-attachments/assets/d020fcd5-bd7e-40b6-a65c-e5574d5c3f88)

**Comment by Ramonlaw** [comment]
i updated to 3.3.4 and sadly doesn't work...

but

![image](https://github.com/user-attachments/assets/08165de3-9975-47f9-a9f1-12b08b11f82c)

i download the Chrome Webstore (v3) export from old extension and import on the Chrome Webstore (v3) one and it works beeps on both now

![image](https://github.com/user-attachments/assets/360d1cbc-e868-4a37-8b39-0ebc798876f1)


maybe its something that you changed from this 3.2.11 vesion

**Comment by steveseguin** [comment]
It's possible although I'm not yet sure what I changed that would have broken it.  Thank you for letting me know though.

**Comment by steveseguin** [comment]
I pushed an update.  If you do want to try the other version again, please let me know how it goes.

I think with either certain twitch-chat settings, or perhaps with a slow-loading twitch chat, it may fail to capture the chat.  I fixed that potential issue.
